skunkworks_crow/src/main/java/org/odk/share/preferences/SettingsPreference.java

```diff
@@ -159,4 +155,25 @@ private void showPasswordDialog() {
         AlertDialog alertDialog = builder.create();
         alertDialog.show();
     }
+
+    public void chooseDirectory() {
```

Comment:
@Chromicle Have you tried handling when there is no intent to handle directory picker?

Reply:
Yeah, I tried but it does not give better results alike as putting intent 
But I checked in Android 6,7,8,9 phones whether the intent is working or not it is perfectly working

Reply:
I tried this in Android 4.4 version and I was not able to select any directory. Can you please verify or can you think of any fallback mechanism?

Reply:
Since there is no "default file manager" in KitKat it is not working if the user is installed a particular 3rd party file manager then it may works

Reply:
@Chromicle If we are adding some feature then we should be confident enough that it won't break anything which was working previously.  So adding this feature will not work as well as won't allow user to manually write the directory path. So I would suggest you rather than printing log or toast it would be beneficial to have some fallback mechanism for some devices in which directory picker is not available.
One more suggestion I would like to give that whenever you're doing some kind of testing then always try to test on less than 5.